_includes/components/header--extended.html

```diff
@@ -68,10 +68,10 @@
             <button class="usa-header-search-button js-search-button">Search</button>
           </li>
           {% endif %}
-          {% assign secondary_links = site.data.navigation[_secondary.links] | default: _secondary.links %}
+          {% assign secondary_links = site.data.navigation[_secondary.links] | default: secondary.links %}
```

Comment:
`secondary` is not defined anywhere so it will always return empty. I _think_ the original intent of this is to allow someone to set a key that accesses an array in `site.data.navigation.something` or to be able to just place an array of links directly under `site.data.header.secondary`.

I think sticking with `_secondary.links` for the default is correct. The simplest solution for removing the extra pipe might just be to unset the default pointer key from `header.yml`. This:
```
secondary:
  # this is a key into _data/navigation.yml
  links: secondary
```
would become:
```
secondary:
  # this is a key into _data/navigation.yml
  links:
```
What's happening now is it's looking to `site.data.navigation.secondary`, seeing that it's empty,  so defaulting to `site.data.header.secondary` - which is just the string `secondary` so when it tries to loop through it it create a blank link that causes the pipe.

I took me a while to really understand how this was working, we could pair on it real quick to if it would be helpful to walkthrough it because it definitely wasn't that intuitive to me at first glance.

Reply:
Ah OK - I see what you mean now - that is a bit confusing to understand. I'm wondering if there is a better way to set up the navigation? Maybe we could pair on this next week? @hursey013

Reply:
yeah, sorry for not following up on pairing this week, i think this could definitely use some refinement though, maybe even if it's just some notes or documentation since it is sort of hard to get your head around.

i also setup a branch locally that checks that the `_secondary` value is indeed an array of links which i think is also an option - i can show you how that works next week.